src/Graph.ts

```diff
@@ -73,6 +76,7 @@ export default class Graph {
 
 	constructor(private readonly options: NormalizedInputOptions, watcher: RollupWatcher | null) {
 		if (options.cache !== false) {
+			this.useCache = true;
```

Comment:
I actually tried that initially but because we `delete options.cache` [here](https://github.com/rollup/rollup/blob/b86e7062424bf8ac3c8d8dea24cb61b7c3c874f2/src/rollup/rollup.ts#L55), it was always `undefined` when we try to access it later.

So I added `useCache`. Unless we could also conditionally `delete` if it's an object only, but that makes the value a bit murky/overloaded, but I'm open to do that if you prefer it too.

Reply:
Ah I see, forgot about that. Maybe instead of deleting we could set the value explicitly to `null` if it is truthy? That would avoid the change in shape of the options object and would correspond to the allowed values of the `cache` option where undefined/null means "generate the cache" and only false disables this.

Reply:
Thinking about it more, I think it makes sense to replace as `undefined` if it's truthy so I've updated that in a468fa3 (`null` isn't in the types). Also removed `useCache`.